airavata-services/profile-service/profile-service-server/src/main/java/org/apache/airavata/service/profile/handlers/TenantProfileServiceHandler.java

```diff
@@ -183,12 +185,25 @@ public boolean deleteGateway(AuthzToken authzToken, String gatewayId) throws Ten
     public boolean isGatewayExist(AuthzToken authzToken, String gatewayId) throws TenantProfileServiceException, AuthorizationException, TException {
         try {
             Gateway gateway = tenantProfileRepository.getGateway(gatewayId);
-            return gateway != null;
+            boolean checkStatus = gateway.getGatewayApprovalStatus().equals(GatewayApprovalStatus.APPROVED);
+            return (gateway != null && checkStatus);
         } catch (Exception ex) {
             logger.error("Error checking if gateway-profile exists, reason: " + ex.getMessage(), ex);
             TenantProfileServiceException exception = new TenantProfileServiceException();
             exception.setMessage("Error checking if gateway-profile exists, reason: " + ex.getMessage());
             throw exception;
         }
     }
+
+    private boolean checkDuplicate(Gateway gateway) throws TenantProfileServiceException {
+        try {
+            Gateway duplicateGateway = tenantProfileRepository.getGateway(gateway.getGatewayId());
```

Comment:
Same concern here as above. This will work only if getGateway returns an APPROVED gateway with this gatewayId.